unlock-app/src/components/interface/checkout/main/checkoutMachine.ts

```diff
@@ -47,6 +47,11 @@ export interface SelectLockEvent {
   skipRecipient?: boolean
   recipients?: string[]
   keyManagers?: string[]
+  hook?: {
+    isCaptcha?: boolean
+    isPassword?: boolean
+    isPromo?: boolean
+  }
```

Comment:
Do we really to make a boolean state?

Our system only supports one hook at a time so add it as a string.

hook = password | promocode | captcha